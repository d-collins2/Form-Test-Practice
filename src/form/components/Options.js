import React from 'react';

const Options = ({options}) => {
    return (
        options && options.map((type, id) => {
            return <option key={id} value={type.display_name}>{type.display_name}</option>
        })
    )
}

Options.defaultProps = {
    options: []
}

export default Options;
