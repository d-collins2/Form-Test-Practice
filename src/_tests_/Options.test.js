import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Options from '../form/Options.js';

const options = ['1', '2', '3', '4'];
describe('<Options />', function(){
    it('should render correctly', function(){
        const component = renderer.create(<Options options={options}/>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should have the correct number of option tags', function(){
        const component = mount(<Options options={options} />);
        expect(component.find('option').length).toEqual(4);
    });

})
