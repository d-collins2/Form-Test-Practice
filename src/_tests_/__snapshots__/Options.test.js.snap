// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Options /> should render correctly 1`] = `
Array [
  <option />,
  <option />,
  <option />,
  <option />,
]
`;
